storage/remote/write_handler_test.go

```diff
@@ -87,8 +291,58 @@ func TestRemoteWriteHandler(t *testing.T) {
 	}
 }
 
-func TestOutOfOrderSample(t *testing.T) {
-	tests := []struct {
+func TestRemoteWriteHandler_V2Message(t *testing.T) {
+	payload, _, _, err := buildV2WriteRequest(log.NewNopLogger(), writeV2RequestFixture.Timeseries, writeV2RequestFixture.Symbols, nil, nil, nil, "snappy")
+	require.NoError(t, err)
+
+	req, err := http.NewRequest("", "", bytes.NewReader(payload))
+	require.NoError(t, err)
+
+	req.Header.Set("Content-Type", remoteWriteContentTypeHeaders[config.RemoteWriteProtoMsgV2])
+	req.Header.Set("Content-Encoding", string(SnappyBlockCompression))
+	req.Header.Set(RemoteWriteVersionHeader, RemoteWriteVersion20HeaderValue)
+
+	appendable := &mockAppendable{}
+	handler := NewWriteHandler(log.NewNopLogger(), nil, appendable, []config.RemoteWriteProtoMsg{config.RemoteWriteProtoMsgV2})
+
+	recorder := httptest.NewRecorder()
+	handler.ServeHTTP(recorder, req)
+
+	resp := recorder.Result()
+	require.Equal(t, http.StatusNoContent, resp.StatusCode)
+
+	b := labels.NewScratchBuilder(0)
+	i := 0
+	j := 0
+	k := 0
+	for _, ts := range writeV2RequestFixture.Timeseries {
+		ls := ts.ToLabels(&b, writeV2RequestFixture.Symbols)
+		for _, s := range ts.Samples {
+			requireEqual(t, mockSample{ls, s.Timestamp, s.Value}, appendable.samples[i])
+			i++
+		}
+		for _, e := range ts.Exemplars {
+			exemplarLabels := e.ToExemplar(&b, writeV2RequestFixture.Symbols).Labels
+			requireEqual(t, mockExemplar{ls, exemplarLabels, e.Timestamp, e.Value}, appendable.exemplars[j])
+			j++
+		}
+		for _, hp := range ts.Histograms {
+			if hp.IsFloatHistogram() {
+				fh := hp.ToFloatHistogram()
+				requireEqual(t, mockHistogram{ls, hp.Timestamp, nil, fh}, appendable.histograms[k])
+			} else {
+				h := hp.ToIntHistogram()
+				requireEqual(t, mockHistogram{ls, hp.Timestamp, h, nil}, appendable.histograms[k])
+			}
+			k++
+		}
+
+		// TODO: check for metadata
```

Comment:
I think this is the only relevant test todo

Reply:
Added test! Was worth it, our conversion of types were broken (: